### ISSUE TITLE
test_unicode_record failing

### ISSUE THREAD

**Issue by jaraco** [pypa/wheel#288]
On my workstation (macOS 10.14.4), the master commit (4fddbba) is failing with this test output:

<details>

```
wheel master $ tox                                                                                                                                                             
GLOB sdist-make: /Users/jaraco/code/public/pypa/wheel/setup.py
python inst-nodeps: /Users/jaraco/code/public/pypa/wheel/.tox/.tmp/package/1/wheel-0.33.1.zip
python installed: atomicwrites==1.3.0,attrs==19.1.0,coverage==4.5.3,more-itertools==7.0.0,pluggy==0.9.0,py==1.8.0,pytest==4.4.1,pytest-cov==2.6.1,six==1.12.0
python run-test-pre: PYTHONHASHSEED='2502144293'
python run-test: commands[0] | python -m pytest
============================================================================= test session starts =============================================================================
platform darwin -- Python 3.7.3, pytest-4.4.1, py-1.8.0, pluggy-0.9.0
cachedir: .tox/python/.pytest_cache
rootdir: /Users/jaraco/code/public/pypa/wheel, inifile: setup.cfg, testpaths: tests
plugins: cov-2.6.1
collected 44 items                                                                                                                                                            

tests/test_bdist_wheel.py .F.....                                                                                                                                       [ 15%]
tests/test_metadata.py .                                                                                                                                                [ 18%]
tests/test_pkginfo.py .                                                                                                                                                 [ 20%]
tests/test_tagopt.py ............                                                                                                                                       [ 47%]
tests/test_wheelfile.py ..................                                                                                                                              [ 88%]
tests/cli/test_convert.py ..                                                                                                                                            [ 93%]
tests/cli/test_pack.py ..                                                                                                                                               [ 97%]
tests/cli/test_unpack.py .                                                                                                                                              [100%]

================================================================================== FAILURES ===================================================================================
_____________________________________________________________________________ test_unicode_record _____________________________________________________________________________

wheel_paths = ['/private/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-26/dist0/complex_dist-0.1-py3-none-...te/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pytest-of-jaraco/pytest-26/dist0/unicode.dist-0.1-py3-none-any.whl']

    def test_unicode_record(wheel_paths):
        path = next(path for path in wheel_paths if 'unicode.dist' in path)
        with ZipFile(path) as zf:
            record = zf.read('unicode.dist-0.1.dist-info/RECORD')
    
>       assert u'åäö_日本語.py'.encode('utf-8') in record
E       AssertionError: assert b'\xc3\xa5\xc3\xa4\xc3\xb6_\xe6\x97\xa5\xe6\x9c\xac\xe8\xaa\x9e.py' in b'unicodedist/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0\nunicodedist/a\xcc\x8aa\xcc\x88o\xcc\x8...1.dist-info/top_level.txt,sha256=TWUW-1C-s7MVSbWlP-D3wzHwgYVuDUpfub8Ex8NEOEA,12\nunicode.dist-0.1.dist-info/RECORD,,\n'
E        +  where b'\xc3\xa5\xc3\xa4\xc3\xb6_\xe6\x97\xa5\xe6\x9c\xac\xe8\xaa\x9e.py' = <built-in method encode of str object at 0x10fb046c0>('utf-8')
E        +    where <built-in method encode of str object at 0x10fb046c0> = 'åäö_日本語.py'.encode

tests/test_bdist_wheel.py:52: AssertionError

---------- coverage: platform darwin, python 3.7.3-final-0 -----------
Name                                                            Stmts   Miss  Cover   Missing
---------------------------------------------------------------------------------------------
.tox/python/lib/python3.7/site-packages/wheel/__init__.py           1      0   100%
.tox/python/lib/python3.7/site-packages/wheel/__main__.py          10     10     0%   5-19
.tox/python/lib/python3.7/site-packages/wheel/bdist_wheel.py      213     25    88%   122, 134, 153-155, 170-171, 219-220, 235, 288-291, 333-341, 345-347
.tox/python/lib/python3.7/site-packages/wheel/cli/__init__.py      56     43    23%   13-16, 24-25, 29-30, 34-35, 39-40, 44-73, 77-88
.tox/python/lib/python3.7/site-packages/wheel/cli/convert.py      157    157     0%   1-269
.tox/python/lib/python3.7/site-packages/wheel/cli/pack.py          31     31     0%   1-58
.tox/python/lib/python3.7/site-packages/wheel/cli/unpack.py        12     12     0%   1-25
.tox/python/lib/python3.7/site-packages/wheel/metadata.py          80      7    91%   22, 26, 29, 39, 55, 61, 106
.tox/python/lib/python3.7/site-packages/wheel/pep425tags.py       110     29    74%   18-20, 27, 29, 31, 35, 42, 51, 60-66, 75-93, 96-99, 109
.tox/python/lib/python3.7/site-packages/wheel/pkginfo.py           29     14    52%   8, 13-25, 30-32
.tox/python/lib/python3.7/site-packages/wheel/util.py              29     10    66%   7-14, 33-34, 39, 46
.tox/python/lib/python3.7/site-packages/wheel/wheelfile.py        111     35    68%   41, 51-77, 82-90, 96-103, 119
wheel/__init__.py                                                   1      0   100%
wheel/__main__.py                                                  10     10     0%   5-19
wheel/bdist_wheel.py                                              213    103    52%   36, 40, 106-134, 139-143, 150-155, 160, 165-180, 184-263, 272, 288-291, 302-304, 333-341, 345-347
wheel/cli/__init__.py                                              56     41    27%   15-16, 24-25, 29-30, 34-35, 39-40, 44-73, 77-88
wheel/cli/convert.py                                              157    101    36%   31-34, 41, 51-56, 71, 78-79, 124-159, 163-249, 261, 264-265, 269
wheel/cli/pack.py                                                  31      3    90%   26, 28, 42
wheel/cli/unpack.py                                                12      0   100%
wheel/metadata.py                                                  80      2    98%   39, 106
wheel/pep425tags.py                                               110     79    28%   18-20, 27, 29, 31, 35, 42, 49-54, 60-66, 72-100, 109, 120-185
wheel/pkginfo.py                                                   29     14    52%   8, 13-25, 30-32
wheel/util.py                                                      29      8    72%   7-14, 39, 46
wheel/wheelfile.py                                                111      1    99%   86
---------------------------------------------------------------------------------------------
TOTAL                                                            1678    735    56%

===================================================================== 1 failed, 43 passed in 8.61 seconds =====================================================================
ERROR: InvocationError for command /Users/jaraco/code/public/pypa/wheel/.tox/python/bin/python -m pytest (exited with code 1)
___________________________________________________________________________________ summary ___________________________________________________________________________________
ERROR:   python: commands failed

```

</details>

**Comment by jaraco** [comment]
Weirdly, the name from the file system and the name in the Python code seem to differ, even though they look the same:

```
(Pdb) os.listdir('tests/testdata/unicode.dist/unicodedist')
['åäö_日本語.py', '__init__.py']
(Pdb) name = os.listdir('tests/testdata/unicode.dist/unicodedist')[0]
(Pdb) name.encode('utf-8')
b'a\xcc\x8aa\xcc\x88o\xcc\x88_\xe6\x97\xa5\xe6\x9c\xac\xe8\xaa\x9e.py'
(Pdb) othername = 'åäö_日本語.py'                
(Pdb) othername.encode('utf-8')
b'\xc3\xa5\xc3\xa4\xc3\xb6_\xe6\x97\xa5\xe6\x9c\xac\xe8\xaa\x9e.py'
```

**Comment by jaraco** [comment]
In the filename as loaded from the file system, what you see is not what's there. The `å` character is actually two: `a` and `\xcc\x8a` (also known as [combining ring above](https://www.fileformat.info/info/unicode/char/030a/index.htm)).

```
(Pdb) name[0]
'a'
(Pdb) name[1]
'̊'
```

But the python name in the same position has the [latin small letter a with a ring above](http://www.fileformat.info/info/unicode/char/e5/index.htm).

Now the question is - whose fault is this? Is it macOS? git?

**Comment by jaraco** [comment]
I was able to create that file on my file system using the direct characters:

```
wheel master $ touch åäö_日本語.py                                                                                                                                             
wheel master $ import os                                                                                                                                                       
wheel master $ os.listdir('.')                                                                                                                                                 
['åäö_日本語.py',
 '.tox',
...
 '.travis.yml']
wheel master $ os.listdir('.')[0][0]                                                                                                                                           
'å'
```

**Comment by jaraco** [comment]
So I'm learning about [unicode normalization](https://en.wikipedia.org/wiki/Unicode_equivalence#Normalization).

**Comment by jaraco** [comment]
I see this issue was observed in [this job](https://travis-ci.com/pypa/wheel/jobs/176929140), part of the `travistest` branch.

**Comment by jaraco** [comment]
Annoyingly, I recreated the file on my local file system using the single-character form, and the bug is gone (tests are passing), but git doesn't detect any change, so there's nothing to commit.

**Comment by jaraco** [comment]
Okay, now this is even weirder. I just wiped and re-cloned the repo. And tests are still passing. So I'm beginning to suspect that there was an issue with a (slighly older) version of git, one which Travis used on the aforementioned job, and one which I was using when I cloned wheel weeks or months ago.

[This release note](https://github.blog/2019-02-24-highlights-from-git-2-21/#detecting-case-insensitive-path-collisions) suggests some changes were recently relating to unicode normalization, even though I can't find related notes in the [official release notes](https://github.com/git/git/blob/master/Documentation/RelNotes/2.20.0.txt). Here's [a case of a similar issue](https://trac.edgewall.org/ticket/12020). The [core.precomposeUnicode option](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreprecomposeUnicode) seems pertinent.
